### DISCUSSION TITLE
Supressing PandaTools use of "NaN" in WriteSDF?

### DISCUSSION THREAD

**Discussion by bgobbi** [rdkit/rdkit#4007] [Q&A]
When reading and writing an sdf file that contains optional tags with PandasTools the missing values are represented as "NaN" and the output file is written with these NaN's. These are realy disruptive as most programs are nto able to deal with NaN. Is there an easy workaround?
This is easy to reproduce with this code:
```
from rdkit.Chem import PandasTools
inFile = "t.sdf"
outFile  = "to.sdf"
pdDoc = PandasTools.LoadSDF(inFile)
PandasTools.WriteSDF(pdDoc,outFile, properties=pdDoc.columns)
```

and this input file:
```
t.sdf
  ChemDraw04052117542D

  1  0  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
M  END
> <a>
1

$$$$
t.sdf
  ChemDraw04052117542D

  1  0  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
M  END
> <b>
2

$$$$
```

**Comment by greglandrum** [comment]
Hi @bgobbi,
Pandas DataFrames have a method for dealing with this that you can use:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.fillna.html
